### PR TITLE
Allow releasing milestones from `master`, create release branches if they don't exist

### PR DESCRIPTION
**Pull request by Ocramius**
This patch fixes some behavior that was quite annoying when dealing with the `roave/*` packages, which I've just configured to use this tool (with a separate environment, using a private GPG subkey by @asgrim).

Specifically, imagine this scenario (also common to the `doctrine/*` packages):

 * `1.1.x` branch
 * `master` branch
 * `1.2.0` milestone

With the setup above, closing milestone `1.2.0` will lead to a crash of the tool, because it cannot find an `1.2.x` branch where to create the tag from.

This patch fixes that. In the scenario above:

 * it will create tag `1.2.0` from `master`
 * it will create an `1.2.x` branch

This seems to be very much compatible with our workflow, since (besides deprecated `Version::` constants) we can safely release new minors from `master`, if BC compliance matches.

Note that this will also allow us to release `2.0.0` from `master`, which previously had to be done manually.

@alcaeus to you specifically (since you asked) I don't think making the tool configurable is a good approach, since it should be opinionated and matching our own workflow (instead of becoming a tool we maintain for the community). If this isn't an approach that applies to `doctrine/*` (I think it is, but @doctrine/doctrinecore has to decide), I'm OK with maintaining my fork of it for `roave/*`.

### CODE REVIEW

File: feature/automated-releases.feature [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+      | name   |
+      | 1.1.x  |
+      | 1.2.x  |
+      | master |
+    And following open milestones:
+      | name  |
+      | 1.1.1 |
+    When I close milestone "1.1.1"
+    Then tag "1.1.1" should have been created on branch "1.1.x"
+    And a new pull request from branch "1.1.x" to "1.2.x" should have been created
```
Comment (alcaeus) [inline feature/automated-releases.feature:71]: I suggest the following scenarios:

```
  Scenario: If a minor release branch doesn't exist, the tool refuses to create it if a newer one exists
    Given following existing branches:
      | name   |
      | 1.2.x  |
      | master |
    And following open milestones:
      | name  |
      | 1.1.0 |
    When I close milestone "1.1.0"
    Then the tool should have halted with an error

  Scenario: If a minor release branch doesn't exist, the tool refuses to create a patch release
    Given following existing branches:
      | name   |
      | master |
    And following open milestones:
      | name  |
      | 1.1.1 |
    When I close milestone "1.1.1"
    Then the tool should have halted with an error
```

File: README.md [hunk 2]
```diff
@@ -1,17 +1,10 @@
...
-7. bot opens PR for 1.2.3-merge-up-to-1.3.x
-
+Please read the [`feature/`](./feature) specification to understand how
+this tool behaves.
+
+TL;DR: when you close a milestone:
+ * a `git tag` will be created
+ * a github release will be created
+ * if not already existing, a release branch will be created
+ * a new pull request will be created, porting changes to the next release branch
```
Comment (alcaeus) [inline README.md]:
```suggestion
 * a new pull request will be created, porting changes to the next release branch from an intermediary branch (e.g. 1.2.3-merge-up-to-1.3.x)
```

**Review by alcaeus — COMMENTED** [review]
Nice 👍

I've suggested a couple of scenarios to test around edge cases:
1. Releasing 1.1.0 after 1.2.x has been released should be prohibited by the tool
2. Releasing 1.1.1 if 1.1.x does not exist should be prohibited by the tool

File: feature/automated-releases.feature (hunk 1 above)
Comment (Ocramius) [reply feature/automated-releases.feature:71]: Hmm, I think that isn't the *current* behavior of the tool, and a good edge case to be prevented.

File: README.md (hunk 2 above)
Comment (Ocramius) [reply README.md]: Included (post-rebase, sorry!) via 9f60354347bdf42036c933c65b17466784863756

Comment (alcaeus) [reply README.md]: > post-rebase

Don’t worry, I don’t stab people for rebasing during code review 😉

File: feature/automated-releases.feature (hunk 1 above)
Comment (Ocramius) [reply feature/automated-releases.feature:71]: Added both scenarios (and tests for them): thanks for the careful review!

**Review by alcaeus — APPROVED** [review]
Great work!
